readthedocs/core/signals.py

```diff
@@ -33,6 +36,54 @@
 post_collectstatic = Signal()
 
 
+@receiver(email_confirmed)
+def process_email_confirmed(request, email_address, **kwargs):
+    """
+    Steps to take after a users email address is confirmed.
+
+    We currently:
+
+    * Subscribe the user to the mailing list if they set this in their signup.
+    * Add them to a drip campaign for new users in the new user group.
+    """
+
+    # email_address is an object
+    # https://github.com/pennersr/django-allauth/blob/6315e25/allauth/account/models.py#L15
+    user = email_address.user
+    profile = UserProfile.objects.filter(user=user).first()
+    if profile and profile.mailing_list:
+        log.bind(
+            email=email_address.email,
+            username=user.username,
+        )
+        log.info("Subscribing user to newsletter and onboarding group.")
+
+        # Try subscribing used to a group, then fallback to normal subscription API
+        url = settings.MAILERLITE_API_ONBOARDING_GROUP_URL
+        if not url:
+            url = settings.MAILERLITE_API_SUBSCRIBERS_URL
+
+        payload = {
+            "email": email_address.email,
+            "resubscribe": True,
+        }
+        headers = {
+            "X-MailerLite-ApiKey": settings.MAILERLITE_API_KEY,
+        }
+        try:
+            resp = requests.post(
+                url,
+                json=payload,
+                headers=headers,
+                timeout=3,  # seconds
+            )
+            resp.raise_for_status()
```

Comment:
Shouldn't this be done in a Celery task?

Reply:
Probably. I was just copying the existing logic for now 🙃. There's definitely more work to be done here, including handling the updating of this on save to unsub the user.

Reply:
I added a TODO note for now.